gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -1718,6 +1747,80 @@ public ListPage<GoogleCloudStorageItemInfo> listObjectInfoPage(
     return new ListPage<>(objectInfos, nextPageToken);
   }
 
+  /**
+   * @see GoogleCloudStorage#listFolderInfoForPrefixPage(String, String, ListFolderOptions, String)
+   */
+  @Override
+  public ListPage<FolderInfo> listFolderInfoForPrefixPage(
+      String bucketName,
+      String objectNamePrefix,
+      ListFolderOptions listFolderOptions,
+      String pageToken)
+      throws IOException {
+    logger.atFiner().log(
+        "listFolderInfoForPrefixPage(%s, %s, %s, %s)",
+        bucketName, objectNamePrefix, listFolderOptions, pageToken);
+
+    ListFoldersRequest.Builder listFoldersRequest =
+        createFolderListRequest(bucketName, objectNamePrefix, listFolderOptions, pageToken);
+
+    if (!isNullOrEmpty(pageToken)) {
+      logger.atFiner().log("listFolderInfoForPrefixPage: next page %s", pageToken);
+      listFoldersRequest.setPageToken(pageToken);
+    }
+
+    List<FolderInfo> listedFolders = new LinkedList<>();
+    String nextPageToken =
+        listStorageFoldersAndPrefixesPage(listFoldersRequest.build(), listedFolders);
+    while (!isNullOrEmpty(nextPageToken)) {
+      nextPageToken =
+          listStorageFoldersAndPrefixesPage(
+              listFoldersRequest.setPageToken(nextPageToken).build(), listedFolders);
+    }
+
+    return new ListPage<>(listedFolders, nextPageToken);
+  }
+
+  private ListFoldersRequest.Builder createFolderListRequest(
+      String bucketName,
+      String objectNamePrefix,
+      ListFolderOptions listFolderOptions,
+      String pageToken) {
+    logger.atFiner().log(
+        "createListFolderRequest(%s, %s, %s, %d)",
+        bucketName, objectNamePrefix, listFolderOptions, pageToken);
```

Comment:
Do we log pageToken at other places? This might contain auth information.

Reply:
Yes pageToken is logged at other places. Please see https://github.com/GoogleCloudDataproc/hadoop-connectors/pull/1153/files#diff-122b4a11389e8967ef7004ddf049fd1e7d688cf1c52fc16e61b0af2e60ed120dR1735